### ISSUE TITLE
Support for sea801 radiator valves

### ISSUE THREAD

**Issue by bikefright** [Yonsm/.homeassistant#5]
Hello Yonsm,

I’m sorry that I don’t speak your language!

I have recently purchased 16 x sea801 smart radiator valves and 3 x sea801-app gateways. I use the saswell app on my phone to control the temperatures.

I also have hassbian with both a z-wave and a ZigBee usb stick but have had no luck in pairing the radiator valves with hassbian, even though they are advertised as operating on 868MHz with 100m range.

I came across your saswell.py code and can see that you are taking a different approach to integrating saswell climate devices into home assistant by accessing the scinan api.

Is there any chance that you would be able to modify your code to support these radiator valves and pull information about them from api.scinan.com into home assistant and therefore be able to control them by pushing settings from home assistant out to api.scinan.com?

Kind regards,
Rowland.

**Comment by bikefright** [comment]
Hello Yonsm,No problem regarding the late reply - we are all busy people 😀.Here in the UK, the Saswell thermostatic radiator valves and RF-to-WiFi gateway are rebranded and sold by a company called Curv360. They have their own app in the Apple App Store, but I am currently using the original Saswell version of the iOS app and it works perfectly.There is an iOS/Android app that communicates with api.scinan.com, and the gateway talks MQTT to/from api.scinan.com.I have control of the radiator valves using the app. I can set the target temperature, see the current reported temperature, switch the radiator valves on (fully open) or off (fully closed), switch to away mode (sets the valve to 16 degrees Celsius). Also for each gateway that I have, I can set all valves (up to 8 valves per gateway) on a single gateway to away mode.I know how to perform dns diversion - I do this for a couple of specific websites already on my pfSense firewall router, so that’s an interesting option for the future 😉.I already have some detailed packet captures that I’ve been looking at in wireshark which contain the entire MQTT payloads between api.scinan.com and one of my saswell gateways. Would you like me to share those with you?  I suspect you will need me to annotate them so that you know which payload corresponds to which activity in the app.  I’ve pretty much decoded the payloads, but have no idea how to code the reading and writing of data to the api 😒.All the best from the UK,Rowland (bikefright).---- On Thu, 13 Dec 2018 09:26:21 +0000  Yonsm<notifications@github.com> wrote ----Hi bikefright, Sorry for my later response:( To fulfill your requirements, I must know what you do on the sea801-app and the corresponding network API data. If you know how to sniffer this data, you can post it up, and I will analyze and support these devices. In addition, the communication between saswell device and api.scinan.com is MQTT protocol. If you can do DNS hijacking, there is actually a way to intercept this data to your MQTT broker and then use MQTT to access the home assistant. This is called local polling mode (no Internet required). Maybe I will consider supporting this method in the future.  —You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.

**Comment by bikefright** [comment]
I've attached some wireshark traces that I've added comments to (look at the Packet Comments) when performing various activities.  Let me know if you need more data or more explanation and I will gladly provide to you.

ATB,
bikefright

[saswell trv wireshark traces.zip](https://github.com/Yonsm/HAExtra/files/2717251/saswell.trv.wireshark.traces.zip)

**Comment by geduxas** [comment]
Hi, bikefright, Yonsm
I am maby second on same boat :) i have couple SEA801 valves and gateway do you have any solution for that?

**Comment by bikefright** [comment]
Sorry geduxas, I abandoned the use of these valves and their gateways in January this year as they were ridiculously unreliable - luckily I was able to return them to Amazon and get a full refund!

I am now using some z-wave valves made by eurotronic (spirit z-wave plus) which have been faultless.  I have 16 of them.  Have a read here of my experience over on the home assistant forum.....

[https://community.home-assistant.io/t/q-add-smart-trvs-to-hive-multizone-or-start-from-scratch-a-deployed-eurotronic-spirit-z-wave-trvs/88920](url)

**Comment by geduxas** [comment]
could you tell me more about problems with saswell, i am curious whats wrong with them? I have searched long time for possible smart valve's and came to saswell products.. maybe i should return them too if i will stuck to same problems ? :)

**Comment by bikefright** [comment]
There are a number of operational problems with the valves, most of which I highlight in my post linked to in my reply above.

Here is a list of the shortfalls IMHO:

1. They are very noisy in operation.

2. They perform a maintenance operation (full open and full close) in the early hours of the morning which in unconfigurable and coupled with the fact they are noisy, this means you cannot put them in a bedroom or people will wake up!
 
3. They randomly lose access to the gateway which means they are no longer controllable remotely.

4. The schedule is difficult to set if you want a change in target temperature ver the midnight boundary......again for full details look at my post on the HA forum linked in my previous reply.
 
5. The schedule is contained on the TRV, so again, not configurable remotely.
 
6. One of the biggest drawbacks is that the TRVs I had were neither Zigbee nor were they Z-Wave standard - they were some other proprietary protocol.

7. There is no native access to the api, so you cannot drive them from a home automation hub like HomeAssistant or OpenHab etc.  This also means that you can never own your own data as you are completely at the mercy of the Saswell cloud platform (which had several outages during my brief use of their valves) where your data resides.
 
8. Because the phone app talks to the saswell cloud and the gateways talk to the saswell cloud (the app does not talk directly to the gateways), if the cloud is down, you have no control over your valves other than walking up to each of them and pressing buttons locally on each valve!

9. They don’t not report valve open percentage.

10. The TRVs randomly lock up and become unresponsive.  Pulling the batteries is the only way to reboot the TRV firmware which is not ideal if you are at work and the valves are stuck in the closed position as you then come home to a cold house/room.

I could go on, but in essence, my experience with them was truly terrible!

Are you looking to integrate some TRVs with HomeAssistant? If so, IMHO, you would be better off returning the saswell valves and gateways and purchasing a cheap Z-Wave USB stick off eBay plus a bunch of eurotronic spirit z-wave TRVs.  In fact ANYTHING would be better 😉.

As I point out in my write up on the HA forum, the only downside of these new eurotronic valves is that you cannot use them without a Z-Wave controller (Z-Wave USB stick). But a £9 usb stick solves that.

Whereabouts in the world are you? Where did you buy the valves from?

ATB,

Bikefright.

**Comment by bikefright** [comment]
Apologies - the link to the Home Assistant community forum did not work before - I’ve corrected this now.

**Comment by geduxas** [comment]
Found your review, after reading it i have returned all stuff i have bought :D for 3 month i tried to find out any of information about saswell TRV's didn't find anything useful.. after i have got gateway, and analysed it via wireshark to understand how it work and is it possible to use for smart home project and accidentally found this post. Thanks for that! You have saved a lot of time to me. 

I am from Lithuania, bought TRV's from local suppliers (16€ each) i think now I know why it was so cheap :)

**Comment by bikefright** [comment]
Yes - I know what you mean regarding the price! mine were on offer for £20/€22 on Amazon :-)

The cheapest I've found the eurotronic spirit TRVs is around the £37/€40 mark from here:

https://www.reichelt.com/gb/en/radiator-thermostat-spirit-z-wave-plus-et-spirit-zwave-p213320.html?r=1

When I bought my everspring sa413 (I actually bought two as I have a test set-up running in parallel) USB sticks they were around £5 - they rose to £9 when I did my write-up on the HA community forum, but I can't find any at the moment.  An excellent alternative is the zwave.me USB stick which can be had for around £20 on ebay second hand.  You'd need to clear it out with OZWCP or similar to get rid of the previous owner's config (if there was any) but an excellent stick as I understand none the less.

If you decide to go the same route as me with HomeAssistant+Z-Wave+eurotronic spirit TRVs, feel free to reach out onthe HA forum as I suspect Yonsm will not appreciate our lengthy debate on his github repo ;-)

ATB,
Bikefright

**Comment by Yonsm** [comment]
Sorry I forgot this issue...For more discussions, you two are welcome here:)

I would be happy if I can provide any  assistance.

**Comment by mickey50** [comment]
我买了一套5个 sea801 和 sea801-app的组合，参考您的代码，我得到了与司南通讯的各种控制api，现在我通过http api可以控制我这套设备，理论上已经可以通过这种方式接入ha了，但是和上面的外国朋友提到的一样，我也想能够通过本地控制的方式来实现接入ha，我用tcpdump抓取了通讯包，但是只发现了网关到司南的mqtt协议，是publish 温度状态的，没有发现司南到网关的控制协议是怎么通讯的，请教如何才能抓取到这部分协议呢，谢谢！

I bought a set of 5 sea801 and sea801-app. Referring to your code, I got various control APIs with Sinan. Now I can control my device through http api.  but like the foreign friends mentioned above, I also want to be able to control them beyond saswell cloud. I used tcpdump to grab the communication package, but only found the packages  from  gateway to Sinan. The mqtt data is publishing temperature state. I not found how the control packages from Sinan to the gateway . How can I get this part of the protocol, thank you!

**Comment by mickey50** [comment]
用笔记本做热点，已经可以抓到双向的mqtt包，只是让人无语的是，所有的信息都是在mqtt的topic里面，居然不用payload。。。

Using a notebook as a hotspot, I can already capture the mqtt contr package, but I found all the message  is in the mqtt topic, not  in the payload
